cpp/include/cudf/table/row_operators.cuh

```diff
@@ -345,6 +345,41 @@ class row_lexicographic_comparator {
     CUDF_EXPECTS(_lhs.num_columns() == _rhs.num_columns(), "Mismatched number of columns.");
   }
 
+  inline __device__ weak_ordering struct_compare(column_device_view const& lhs,
+                                                 column_device_view const& rhs,
+                                                 null_order null_precedence,
+                                                 size_type lhs_index,
+                                                 size_type rhs_index) const noexcept
+  {
+    if (has_nulls) {
+      bool const lhs_is_null{lhs.nullable() and lhs.is_null(lhs_index)};
+      bool const rhs_is_null{rhs.nullable() and rhs.is_null(rhs_index)};
+
+      if (lhs_is_null and rhs_is_null) {  // null <? null
+        return weak_ordering::EQUIVALENT;
+      } else if (lhs_is_null) {  // null <? x
+        return (null_precedence == null_order::BEFORE) ? weak_ordering::LESS
+                                                       : weak_ordering::GREATER;
+      } else if (rhs_is_null) {  // x <? null
+        return (null_precedence == null_order::AFTER) ? weak_ordering::LESS
+                                                      : weak_ordering::GREATER;
+      }
+    }
+
+    for (size_type i = 0; i < lhs.num_child_columns(); ++i) {
+      weak_ordering state{weak_ordering::EQUIVALENT};
+
+      auto comparator =
+        element_relational_comparator<has_nulls>{lhs.child(i), rhs.child(i), null_precedence};
+
+      state = cudf::type_dispatcher(lhs.child(i).type(), comparator, lhs_index, rhs_index);
+
+      if (state == weak_ordering::EQUIVALENT) { continue; }
+      return state;  // == (ascending ? weak_ordering::LESS : weak_ordering::GREATER);
+    }
+    return weak_ordering::EQUIVALENT;
```

Comment:
Couldn't this instead just invoke `row_lexicographic_comparator` on the children treating them as a `table`?

Reply:
Tried various combinations of recursion.
Either recursion causes indefinite compilation time or causes ptx warning
`ptxas warning : Stack size for entry function cannot be statically determined`

Reply:
`row_lexicographic_comparator` can't be reused because 
return type is bool,  
`ascending` usage for each columns, which does not exist for struct child columns.

Reply:
Oh, right. This would end up being recursive.